packages/model-viewer/src/features/scene-graph.ts

```diff
@@ -210,8 +204,22 @@ export const SceneGraphMixin = <T extends Constructor<ModelViewerElementBase>>(
           shadow.visible = false;
         }
 
-        const exporter = new GLTFExporter();
-        exporter.parse(scene.modelContainer, (gltf) => {
+        const currentGLTF = this[$currentGLTF];
+
+        if (currentGLTF != null && 'functions' in currentGLTF.userData &&
+            'ensureLoadVariants' in currentGLTF.userData.functions) {
+          // Ensure all variant materials are loaded because some of them may
+          // not be loaded yet.
+          await currentGLTF.userData.functions.ensureLoadVariants(scene);
+        }
+
+        const exporter =
+            new GLTFExporter()
+                // @ts-ignore
```

Comment:
[@types/three](https://www.npmjs.com/package/@types/three) doesn't have `GLTFExporter` plugin API  (register, unregister, GLTFExporterPlugin, GLTFWriter) declaration yet so inserting `// @ts-ignore` and using `any` so far.

Reply:
Ah yes, there's a bunch of updates we should make to those types. I prefer `any` to `// @ts-ignore` because it's a bit more specific about the problem: `new (GLTFExporter() as any).register`.

Reply:
Done